cloudferrylib/os/identity/keystone.py

```diff
@@ -451,13 +455,13 @@ def _passwd_notification(self, email, name, password):
                                               'password': password}))
 
     def _deploy_roles(self, roles):
-        dst_roles = {role.name: role.id for role in self.get_roles_list()}
+        dst_roles = {role.name.lower(): role.id for role in self.get_roles_list()}
         for _role in roles:
             role = _role['role']
-            if role['name'] not in dst_roles:
+            if role['name'].lower() not in dst_roles:
```

Comment:
maybe it would be better to store all the role names in lowercase during `read_info`? At least in this case it will be consistent